src/helpers.ml

```diff
@@ -2,11 +2,26 @@ open Base
 
 let f = Printf.sprintf
 
+let rec fold_string_matches ~regexp ~f ~init ?(pos = 0) string =
+  ( try
+      let _ = Str.search_forward (Str.regexp regexp) string pos in
```

Comment:
```suggestion
      let () = Str.search_forward (Str.regexp regexp) string pos in
```
I think it is good practice not to leave the pattern empty to make sure it is really evaluated.

Reply:
`search_forward` returns an `int`, not a unit.  Also the previous code was already
```
try
    let _ = Str.search_forward (Str.regexp regexp) string 0 in
    true
  with Stdlib.Not_found -> false
```

Reply:
In that case could you do `(_ : int)`.

Reply:
Would
```ocaml
let string_match ~regexp ?(pos = 0) string =
  try
    let _ = Str.search_forward (Str.regexp regexp) string pos in
    true
  with Stdlib.Not_found -> false

let rec fold_string_matches ~regexp ~f ~init ?(pos = 0) string =
  if string_match ~regexp ~pos string then
    let pos = Str.match_end () in
    f (fun () -> fold_string_matches ~regexp ~f ~init ~pos string)
  else
      init
```
be clearer?

Reply:
That would be clearer but my suggestion was to make the code a bit more rigid to changes. If an argument is omitted during refactoring say, then suddenly the search_forward won't work. By adding `(_ : int)` you guarantee that it will have been evaluated.

Not super important anyway, merge if you wish.